e2e/util_test.go

```diff
@@ -267,6 +365,38 @@ func createBucket(t *testing.T, client *s3.S3, bucket string) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	t.Cleanup(func() {
+		// cleanup if bucket exists.
+		_, err := client.HeadBucket(&s3.HeadBucketInput{Bucket: aws.String(bucket)})
+		if err == nil {
+
+			listInput := s3.ListObjectsInput{
+				Bucket: aws.String(bucket),
+			}
+
+			//remove objects inside the bucket first.
+			err := client.ListObjectsPages(&listInput, func(p *s3.ListObjectsOutput, lastPage bool) bool {
+				for _, c := range p.Contents {
+					client.DeleteObject(&s3.DeleteObjectInput{
```

Comment:
You can delete multiple objects at once. Please use that.

Reply:
Unfortunately, GCS does not support multi delete. If we use `DeleteObjects`, then it won't work with gcs.